skimage/morphology/_flood_fill_cy.pyx

```diff
@@ -0,0 +1,143 @@
+
+    with nogil:
+        # Initialize the queue
+        queue_init(&queue, 64)
+        try:
+            queue_push(&queue, &start_index)
+            flags[start_index] = FILL
+            # Break loop if all queued positions were evaluated
+            while queue_pop(&queue, &current_index):
+                # Look at all neighboring samples
+                for i in range(neighbor_offsets.shape[0]):
+                    neighbor = current_index + neighbor_offsets[i]
+
+                    # Shortcut if neighbor is already part of fill
+                    if flags[neighbor] == UNKNOWN:
+                        if image[neighbor] == seed_value:
+                            # Neighbor is in fill; check its neighbors too.
+                            flags[neighbor] = FILL
+                            queue_push(&queue, &neighbor)
+        finally:
+            # Ensure memory released
+            queue_exit(&queue)
+
+
+cpdef inline void _flood_fill_tolerance(dtype_t[::1] image,
+                                        unsigned char[::1] flags,
+                                        Py_ssize_t[::1] neighbor_offsets,
+                                        Py_ssize_t start_index,
+                                        dtype_t seed_value,
+                                        dtype_t low_tol,
+                                        dtype_t high_tol) nogil:
+    """Find connected areas to fill, within a tolerance.
+
+    Parameters
+    ----------
+    image : ndarray, one-dimensional
+        The raveled view of a n-dimensional array.
+    flags : ndarray, one-dimensional
+        An array of flags that is used to store the state of each pixel during
+        evaluation.
+    neighbor_offsets : ndarray
+        A one-dimensional array that contains the offsets to find the
+        connected neighbors for any index in `image`.
+    queue_ptr :
+        Pointer to initialized queue.
+    start_index : int
+        Start position for the flood-fill.
+    seed_value :
+        Value of `image[start_index]`.
+    low_tol :
+        Lower limit for tolerance comparison.
+    high_tol :
+        Upper limit for tolerance comparison.
+    """
+    cdef:
+        QueueWithHistory queue
+        QueueItem current_index, neighbor
+
+    with nogil:
+        # Initialize the queue and push start position
+        queue_init(&queue, 64)
+        try:
+            queue_push(&queue, &start_index)
+            flags[start_index] = FILL
+            # Break loop if all queued positions were evaluated
+            while queue_pop(&queue, &current_index):
+                # Look at all neighboring samples
+                for i in range(neighbor_offsets.shape[0]):
+                    neighbor = current_index + neighbor_offsets[i]
+
+                    # Only do comparisons on points not (yet) part of fill
+                    if flags[neighbor] == UNKNOWN:
+                        if low_tol <= image[neighbor] <= high_tol:
```

Comment:
Is this syntax correct in cython? or is it doing ((low_tol <= image) <= high_tol)

Reply:
@hmaarrfk it's correct. Cython tries very hard to be a superset of Python.

Reply:
ok. 

I'll just leave an other comment that I really don't like this syntax. 
I know that working with lowest common denominators is "bad" but seeing as this causes many bugs in other languages, I think it is a good syntax "trick" to avoid.

Reply:
@hmaarrfk I disagree, I think all other languages are broken in this regard and major advantages of Python should be celebrated. =)

Having said that, I don't *strongly* disagree because, tragically, NumPy arrays don't have this syntax (I guess because Python as far as I know doesn't provide a way to override this behaviour for classes), and *that* has bitten me a couple of times. Not bitten me hard enough that I introduced a bug, but hard enough that I had to rewrite something because I forgot about this not working.

Reply:
Truthfully, my issue is with CPython. The fact that
```
for i in range(10):
    pass
```
can't be optimized is a travesty

Reply:
I don't think this is the right forum for a chat about the Python language's weaknesses? =)

Anyway, "can't" is such a strong word... CPython is FOSS, so perhaps this can be your contribution to the language? ;)